types/virtuals.d.ts

```diff
@@ -0,0 +1,12 @@
+/// <reference path="./utility.d.ts" />
+
+declare module 'mongoose' {
+    type VirtualPathFunctions<DocType = {}, pathType = unknown> = {
+      get?: (this: Document<any, any, DocType> & DocType) => pathType;
+      set?: (this: Document<any, any, DocType> & DocType, ...args: any) => unknown;
```

Comment:
Why is this `...args: any`? We call setters with a well defined function signature here: https://github.com/Automattic/mongoose/blob/0612349cbbd1a442d78a0d2802ce88015cadb6b2/lib/virtualtype.js#L168

Reply:
Do you mean that this type should be a single function or an array of that function? 
OR  is it correct as it is now?

Reply:
@mohammad0-0ahmad sorry I missed your question. The current type definition is correct

Reply:
No problem, it is okay